### PR TITLE
Vue: Add basic CSF types

### PR DESCRIPTION
**Pull request by tooppaaa**
Issue: #11845

## What I did

Add typings for Vue

### CODE REVIEW

File: app/vue/src/client/preview/types-6-0.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Component } from 'vue';
+import { Args as DefaultArgs, Annotations, BaseMeta, BaseStory } from '@storybook/addons';
+import { StoryFnVueReturnType } from './types';
+
+export { Args, ArgTypes, Parameters, StoryContext } from '@storybook/addons';
+
+type VueComponent = any;
```
Comment (shilman) [inline app/vue/src/client/preview/types-6-0.ts:7]: Did you mean `Component`? deepscan is giving a warning about that import

**Review by shilman — COMMENTED** [review]
LGTM! 🙏

File: app/vue/src/client/preview/types-6-0.ts (hunk 1 above)
Comment (tooppaaa) [reply app/vue/src/client/preview/types-6-0.ts:6]: I did meant any (but forgot to cleanup my tests using Component leading to unused import)

This is only a very first step of types for Vue, it's not infering args as I would expect but no idea how to make this working for now. Need Vue experts :)

Comment (shilman) [reply app/vue/src/client/preview/types-6-0.ts:6]: @backbone87 @Aaron-Pool any thoughts on the Component type in Vue?

Comment (pocka) [reply app/vue/src/client/preview/types-6-0.ts:6]: To represent a general component, `Component<any, any, any, any> | AsyncComponent<any, any, any, any>`, which is the typing for `components` property of components, would be good IMO. (or refer it with `ComponentOptions<any>['components']['string']` ...? 🤔 )

https://github.com/vuejs/vue/blob/dev/types/options.d.ts#L102

**Review by shilman — APPROVED** [review]
Looks great! Will wait to merge until we "start" 6.1 next week

**Pull request merged**